src/App.tsx

```diff
@@ -1,7 +1,27 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    currentChar: 'Nothing was pressed yet',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ currentChar: `The last pressed key is [${event.key}]` });
+    });
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', () => {
+      this.setState({ currentChar: 'Nothing was pressed yet' });
+    });
+  }
```

Comment:
move callback outside of event listener cause now you don't actually delete it
https://mate.academy/learn/react/class-components#/video/3221
watch from 5:05